docs/src/chapters/04-wizja.adoc

```diff
@@ -1,9 +1,327 @@
+tekstowych. Tak samo, jak w przypadku dokumentacji pozwala to na trzymanie plików w systemie kontroli wersji.
+
+==== Testy E2E
+
+Cypress::
+Cypress to framework do tworzenia automatycznych testów E2E dla aplikacji webowych. Dostarcza on wszystkich narzędzi
+wymaganych do przeprowadzenia testów. Same testy pisane są w języku JavaScript.
+
+Cucumber::
+Jest to narzędzie programistyczne wspierające proces BDD. Pozwala na uruchamianie testów akceptacyjnych oraz E2E
+zapisanych przy pomocy specjalnego języka DSL Gherkin. Pozwala on na opisanie zachowania aplikacji ze strony biznesu
+bez zagłębiania się w szczegóły implementacji.
 
 === Wykorzystane Narzędzia
+
+Bitcoind::
+Jest to implementacja pełnego węzła sieci Bitcoin, wspierająca wszystkie funkcjonalności protokołu.
+
+LND::
+Jest to kompletna implementacja węzła sieci Lightning, w pełni kompatybilna ze specyfikacją BOLT.
+
+IntelliJ IDEA Ultimate::
+Jest to zintegrowane środowisko programistyczne stworzone początkowo dla ekosystemu Javy. Współcześnie wspiera wiele
+języków, w tym JavaScript oraz TypeScript. Jest głównym narzędziem używanym do prac nad projektem. Wersja _ultimate_
+jest płatna, aczkolwiek cały zespół korzysta z bezpłatnych licencji studenckich zapewnionych przez uczelnię.
+
+Docker::
+Docker jest oprogramowaniem służącym do konteneryzacji. Jest to platforma do uruchamiania i wdrażania aplikacji.
+W projekcie za pomocą Dockera przygotowywane jest środowisko deweloperskie w postaci bazy danych czy też węzłów
+Lightning i Bitcoin.
+
+Figma::
+Figma jest aplikacją webową stworzoną do edycji grafiki wektorowej oraz tworzenia prototypów. Została wykorzystana do
+stworzenia makiety interaktywnej w ramach projektu interfejsu użytkownika.
+
+Maven::
+Maven jest narzędziem do budowania aplikacji napisanych w Javie. Funkcjonalności można rozbudować poprzez rozmaite
+wtyczki. Użyty w projekcie do automatyzacji budowania, testowania oraz pakowania serwera.
+
+Draw.io::
+Jest to aplikacja webowa stworzona do rysowanie wszelkiego rodzaju diagramów. Została użyta do stworzenia diagramu
+architektury systemu.
+
+Make::
+Make jest narzędziem do automatycznego budowania aplikacji i nie tylko. Używany jest w projekcie do budowania
+dokumentacji oraz książki projektowej.
+
+PostgreSQL::
+Jest to system zarządzania relacyjną bazą danych SQL.
+
+Yarn::
+Jest to system budowania i pakowania aplikacji dla aplikacji napisanych w języku JavaScript na platformie _node.js_.
+Wykorzystywany zarówno w aplikacji mobilne, jak i aplikacji webowej.
+
+Git::
+Git jest systemem kontroli wersji stworzonym przez Linusa Torvaldsa.
+
+Github::
+Jest to serwis internetowy hostujący repozytoria dla systemu Git. Został wybrany jako lokalizację dla repozytorium
+projektu.
+
+Github Actions::
+Jeden z podsystemów Github'a pozwalający na tworzenie procesu automatyzacji CI/CD dla repozytoriów.
+
+Discord::
+Discord jest aplikacją służącą do komunikacji głosowej oraz przy pomocy wiadomości tekstowej. Używany jako główne
```

Comment:
troche zmieniłbym to zdanie, np na coś takiego:
"...komunikacji głosowej, wideokonferencji oraz przesyłania wiadomości tesktowych"